### PR TITLE
[CHORE] Improve the performance of the app, reducing calls on main-thread and the number of reutilization of the cells

### LINKED ISSUE #1064: [BUG] Something is blocking the UI of the app on reconnecting to the server

### PR DESCRIPTION
**Pull request by rafaelks**
@RocketChat/ios

Closes #1064 

## Description

This PR fixes a few performance and bugs in the app:

1. ImageView flickering on the messages list;
2. Cell was being reused hundreds of times on any Message update call from Realm;
3. Moved the Markdown processing to the main thread again, to avoid weird behaviors;

### CODE REVIEW

**Comment by rafaelks** [comment]
@cardoso This is ready to review!

File: Rocket.Chat/Managers/Model/AuthSettingsManager.swift [hunk 1]
```diff
@@ -38,7 +38,7 @@ final class AuthSettingsManager {
                 return
             }
 
-            Realm.executeOnMainThread({ realm in
+            Realm.execute({ realm in
                 let settings = AuthManager.isAuthenticated()?.settings ?? AuthSettings()
                 settings.map(response.result["result"], realm: realm)
                 realm.add(settings, update: true)
```
Comment (cardoso) [inline Rocket.Chat/Managers/Model/AuthSettingsManager.swift]: Just crashed here 😦 "not in a write transaction" when changing servers.

Comment (cardoso) [reply Rocket.Chat/Managers/Model/AuthSettingsManager.swift]: Looks like this causes very bad problems @rafaelks, can we revert only this part? Can't even connect to servers with this.

File: Rocket.Chat/Views/Cells/Chat/ChatMessageTextViewModel.swift [hunk 2]
```diff
@@ -43,9 +43,9 @@ final class ChatMessageTextViewModel {
         self.attachment = attachment
     }
 
-    func toggleCollpase() {
-        Realm.executeOnMainThread({ _ in
+    func toggleCollpase(_ completion: VoidCompletion? = nil) {
+        Realm.execute({ _ in
             self.attachment.collapsed = !self.attachment.collapsed
```
Comment (cardoso) [inline Rocket.Chat/Views/Cells/Chat/ChatMessageTextViewModel.swift]: Crashing here too

**Pull request merged**